src/main/java/practice/CandidateValidator.java

```diff
@@ -1,5 +1,27 @@
 package practice;
 
-public class CandidateValidator {
-    //write your code here
+import java.util.function.Predicate;
+import model.Candidate;
+
+public class CandidateValidator implements Predicate<Candidate> {
+    private static final int MIN_YEARS_OLD = 35;
+    private static final int MIN_TOTAL_DURATION = 10;
+    private static final String VALID_NATIONALITY = "Ukrainian";
+    private static final String REGEX = "-";
+    private static final int SINCE_YEAR_INDEX = 0;
+    private static final int TO_YEAR_INDEX = 1;
+
+    @Override
+    public boolean test(Candidate candidate) {
+        if (candidate.getAge() >= MIN_YEARS_OLD && candidate.isAllowedToVote()
+                && candidate.getNationality().equals(VALID_NATIONALITY)) {
```

Comment:
```suggestion
                && VALID_NATIONALITY.equals(candidate.getNationality())) {
```
In that way, you will be sure that you never will get NPE.